### ISSUE TITLE
Fix build error on CI

### ISSUE THREAD

**Issue by sophiabrandt** [zero-to-mastery/ZtM-Job-Board#2987]
Build fails on CI with:
```
+ npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: react-scripts@5.0.0
npm ERR! Found: typescript@5.1.6
npm ERR! node_modules/typescript
npm ERR!   typescript@"^5.1.6" from the root project
npm ERR!   peer typescript@">= 2.7" from fork-ts-checker-webpack-plugin@6.5.0
npm ERR!   node_modules/fork-ts-checker-webpack-plugin
npm ERR!     fork-ts-checker-webpack-plugin@"^6.5.0" from react-dev-utils@12.0.0
npm ERR!     node_modules/react-dev-utils
npm ERR!       react-dev-utils@"^12.0.0" from react-scripts@5.0.0
npm ERR!       node_modules/react-scripts
npm ERR!         react-scripts@"^5.0.0" from the root project
npm ERR!   1 more (tsutils)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional typescript@"^3.2.1 || ^4" from react-scripts@5.0.0
npm ERR! node_modules/react-scripts
npm ERR!   react-scripts@"^5.0.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: typescript@4.9.5
npm ERR! node_modules/typescript
npm ERR!   peerOptional typescript@"^3.2.1 || ^4" from react-scripts@5.0.0
npm ERR!   node_modules/react-scripts
npm ERR!     react-scripts@"^5.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by SimardeepSingh-zsh** [comment]
The error you faced is related to conflicting dependencies in your Node.js project. The main issue is that two packages, "react-scripts" and "typescript", have conflicting requirements for the "typescript" package version.

Try upgrading your TypeScript to a version that satisfies both "react-scripts" and your project requirements. 
"dependencies": {
    "typescript": "^3.2.1"
    // other dependencies
}

Check if there's an updated version of "react-scripts."
"dependencies": {
    "react-scripts": "^5.0.1" // or a version that is compatible with TypeScript 5.1.6
    // other dependencies
}

As suggested in the error message, you can try running the installation command with the --force or --legacy-peer-deps flag. However, this might lead to potential issues with your project's dependencies, so it's not the recommended solution unless you have no other option.
npm install --force
# or
npm install --legacy-peer-deps

If your project has any local dependencies or packages that might have their own conflicting TypeScript versions specified, ensure that those are resolved as well.
If you're still facing issues with npm, you can try using Yarn as an alternative package manager. Yarn sometimes resolves dependency conflicts differently and might work better for your project.

**Comment by sophiabrandt** [comment]
Fixed with #2991
